TestStoreZoneUpdateAndRangeSplit is flaky

The following test appears to have failed:

[#9587](https://circleci.com/gh/cockroachdb/cockroach/9587):

```
W1124 18:54:16.485918 903 storage/store.go:1341  conflicting intents on [/System/Meta2/"\xff\f\x03\xe8ZJfGAbTYxaPiwjCPvCNncIHEEbqsYLsbKTRkzCjHgsQETXLSnMYmnLAnuDesrEhwiTXIIKwtDgamIfhAewgELLRRZutEDtwxmMVK"]: resolved? true
I1124 18:54:16.493725 903 storage/replica_command.go:977  pushing intent from txn with lower priority "storage/replica_command.go:1318 (*Replica).AdminSplit" id=85c0326b key=/Local/Range/RangeDescriptor/"\xff\f\x03\xe8ZJfGAbTYxaPiwjCPvCNncIHEEbqsYLsbKTRkzCjHgsQETXLSnMYmnLAnuDesrEhwiTXIIKwtDgamIfhAewgELLRRZutEDtwxmMVK" rw=false pri=72.17366773 iso=SERIALIZABLE stat=PENDING epo=0 ts=0.000000000,766 orig=0.000000000,766 max=0.000000000,766 vs 1975588288
W1124 18:54:16.496811 903 storage/store.go:1341  conflicting intents on [/Local/Range/RangeDescriptor/"\xff\f\x03\xe8ZJfGAbTYxaPiwjCPvCNncIHEEbqsYLsbKTRkzCjHgsQETXLSnMYmnLAnuDesrEhwiTXIIKwtDgamIfhAewgELLRRZutEDtwxmMVK"]: resolved? true
W1124 18:54:16.503140 903 storage/store.go:1341  conflicting intents on [/Local/Range/RangeDescriptor/"\xff\f\x03\xe8"]: resolved? true
I1124 18:54:16.715820 903 storage/engine/rocksdb.go:132  closing in-memory rocksdb instance
--- FAIL: TestStoreZoneUpdateAndRangeSplit (1.50s)
	client_split_test.go:492: expected range to split within 1s
	client_split_test.go:497: second range has split: range_id:2 start_key:"\377\014\003\350" end_key:"\377\014\003\350ZJfGAbTYxaPiwjCPvCNncIHEEbqsYLsbKTRkzCjHgsQETXLSnMYmnLAnuDesrEhwiTXIIKwtDgamIfhAewgELLRRZutEDtwxmMVK" replicas:<node_id:1 store_id:1 replica_id:1 > next_replica_id:2 
=== RUN   TestStoreRangeSplitWithMaxBytesUpdate
I1124 18:54:16.717557 903 gossip/gossip.go:172  gossiping node descriptor node_id:1 address:<network_field:"localhost" address_field:"1" > attrs:<> 
I1124 18:54:16.722214 903 multiraft/multiraft.go:557  node 1 starting
I1124 18:54:16.723475 903 raft/raft.go:409  [group 1] 1 became follower at term 5
I1124 18:54:16.723700 903 raft/raft.go:218  [group 1] newRaft 1 [peers: [1], term: 5, commit: 10, applied: 10, lastindex: 10, lastterm: 5]
I1124 18:54:16.723983 903 multiraft/multiraft.go:959  node 1 campaigning because initial confstate is [1]
I1124 18:54:16.724504 903 raft/raft.go:489  [group 1] 1 is starting a new election at term 5
I1124 18:54:16.724615 903 raft/raft.go:422  [group 1] 1 became candidate at term 6
--
I1124 18:54:22.780317 903 storage/replica.go:1369  gossiping system config from store 1, range 2
I1124 18:54:22.792994 903 storage/engine/rocksdb.go:132  closing in-memory rocksdb instance
--- PASS: TestStoreSplitReadRace (0.06s)
=== RUN   Example_rebalancing
--- PASS: Example_rebalancing (0.72s)
FAIL
FAIL	github.com/cockroachdb/cockroach/storage	32.900s
=== RUN   TestBatchBasics
I1124 18:54:05.871518 962 storage/engine/rocksdb.go:132  closing in-memory rocksdb instance
--- PASS: TestBatchBasics (0.00s)
=== RUN   TestBatchGet
I1124 18:54:05.875567 962 storage/engine/rocksdb.go:132  closing in-memory rocksdb instance
--- PASS: TestBatchGet (0.01s)
=== RUN   TestBatchMerge
I1124 18:54:05.882461 962 storage/engine/rocksdb.go:132  closing in-memory rocksdb instance
--- PASS: TestBatchMerge (0.01s)
=== RUN   TestBatchProto

```
Please assign, take a look and update the issue accordingly.

--- a/storage/replica_command.go
+++ b/storage/replica_command.go
@@ -916,12 +916,11 @@ func (r *Replica) PushTxn(batch engine.Engine, ms *engine.MVCCStats, h roachpb.H
 		}
 	} else {
 		// The transaction doesn't exist on disk; we're allowed to abort it.
+		// TODO(tschottdorf): for GC purposes, don't always want to write
+		// a new entry, but #3219 shows that we need to do so carefully.
+
 		reply.PusheeTxn = *args.PusheeTxn.Clone()
 		reply.PusheeTxn.Status = roachpb.ABORTED
-		if args.PushType == roachpb.CLEANUP_TXN {
-			// If we're only here to clean up, no reason to persist anything.
-			return reply, nil
-		}
 		return reply, engine.MVCCPutProto(batch, ms, key, roachpb.ZeroTimestamp, nil, &reply.PusheeTxn)
 	}
 

